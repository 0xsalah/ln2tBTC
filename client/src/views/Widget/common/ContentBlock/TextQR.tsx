import React, { useRef, useState, useEffect } from 'react'
import ICONS from '../../../../img/icons.svg'
import { Hint, Tooltip } from '../../common/Tooltip'
import QRCode from 'qrcode'

interface TextQRProps {
    text: string
    hint?: React.ReactNode
}

const TextQR: React.FC<TextQRProps> = ({ text, hint }) => {
    let questionMarkElement = useRef<HTMLElement>(null);
    const [displayTooltip, setDisplayTooltip] = useState(false)

    const [QRUrl, setQRUrl] = useState<string>()

    useEffect(() => {
        const generateQr = async () => {
            const url = await QRCode.toDataURL(text)
            setQRUrl(url)
        }
        generateQr()
    }, [text])

    return (
        <>
            <div className="invoice__block-text">{text}</div>
            <div className="invoice__block-code">
                <Hint
                    ref={questionMarkElement}
                    onClick={() => { setDisplayTooltip(true) }}
                    svgIcon={
                        <svg className="icon icon-code">
                            <use xlinkHref={`${ICONS}#icon-code`}></use>
                        </svg>
                    } />
            </div>
            {
                <Tooltip active={displayTooltip} hintButton={questionMarkElement} onDismiss={() => setDisplayTooltip(false)} >
                    {QRUrl ?
                        <img className="image-code" src={QRUrl} width="185" height="187" alt="" />
                        : <i style={{ textAlign: 'center', width: '155px', height: '157px' }} className="image-code icon icon-loader"></i>
                    }
                </Tooltip>
            }
        </>
    )
}

export default TextQR