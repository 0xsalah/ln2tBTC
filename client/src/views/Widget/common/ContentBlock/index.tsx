import React from 'react'
import { default as TextQR } from './TextQR'

interface ContentBlockProps {
    label?: string
}

const ContentBlock: React.FC<ContentBlockProps> = (props) => {
    const { label, children } = props
    return (
        <div className="invoice__block invoice__block--invoice">
            {label && <div className="invoice__block-title">{label}</div>}
            <div className="invoice__block-content row">
                {children}
            </div>
        </div>
    )
}

export default ContentBlock
export { TextQR }