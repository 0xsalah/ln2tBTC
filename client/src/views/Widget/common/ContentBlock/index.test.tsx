import React from "react";
import { render } from "@testing-library/react";
import ContentBlock from ".";

test("snapshot of component rendered with no label", () => {
  const component = render(
    <ContentBlock>
      <span>Test</span>
    </ContentBlock>
  );
  expect(component.baseElement).toMatchSnapshot();
});

test("snapshot of component rendered with label", () => {
  const component = render(
    <ContentBlock label="test">
      <span>Test</span>
    </ContentBlock>
  );
  expect(component.baseElement).toMatchSnapshot();
});
