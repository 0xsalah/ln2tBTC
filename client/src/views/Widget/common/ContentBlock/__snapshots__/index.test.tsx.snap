// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot of component rendered with label 1`] = `
<body>
  <div>
    <div
      class="invoice__block invoice__block--invoice"
    >
      <div
        class="invoice__block-title"
      >
        test
      </div>
      <div
        class="invoice__block-content row"
      >
        <span>
          Test
        </span>
      </div>
    </div>
  </div>
</body>
`;

exports[`snapshot of component rendered with no label 1`] = `
<body>
  <div>
    <div
      class="invoice__block invoice__block--invoice"
    >
      <div
        class="invoice__block-content row"
      >
        <span>
          Test
        </span>
      </div>
    </div>
  </div>
</body>
`;
